src/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -68,18 +57,27 @@ private int Run(string[] args)
                     }
                     else if (ArgumentMatch(arg, "out") || ShortNameArgumentMatch(arg, "o"))
                     {
-                        if (codePath != null)
+                        if (codePath != null || ++i >= args.Length)
                         {
-                            errs.Add(SR.Format(SR.ErrInvalidArgument, "--out", arg));
+                            if (arg.Length == "-o".Length)
+                                errs.Add(SR.Format(SR.ErrInvalidArgument, "-o", arg));
```

Comment:
Maybe I missed something. I don't see ErrInvalidArgument accepts 2 arguments.

Reply:
@zhenlan You are correct, just fixed that.